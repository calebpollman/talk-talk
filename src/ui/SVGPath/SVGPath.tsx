import { SVGAttributes } from 'react';

interface SVGPathProps extends Omit<SVGAttributes<SVGPathElement>, 'children'> {}

function SVGPath(props: SVGPathProps) {
  return <path {...props} />;
}

export default SVGPath;
