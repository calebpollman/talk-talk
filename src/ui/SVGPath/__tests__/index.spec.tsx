import { ReactNode } from 'react';
import { render } from '@testing-library/react';

import SVGPath from '..';

const mockProps = {
  fill: '#fff',
  d: 'M462.6,193.8c-1.1,1.1-2.2,2.3-4.1,2.1c-2.3-0.1-4.5-0.7-6.1-2.2z',
};

// jest expects the svg path element to be wrapped in an svg
const TestWrapper = ({ children }: { children: ReactNode }) => <svg>{children}</svg>;

describe('SVGPath', () => {
  it('renders as expected', () => {
    const { baseElement } = render(
      <TestWrapper>
        <SVGPath {...mockProps} />
      </TestWrapper>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
