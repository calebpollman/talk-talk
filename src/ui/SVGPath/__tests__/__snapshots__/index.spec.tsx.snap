// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVGPath renders as expected 1`] = `
<body>
  <div>
    <svg>
      <path
        d="M462.6,193.8c-1.1,1.1-2.2,2.3-4.1,2.1c-2.3-0.1-4.5-0.7-6.1-2.2z"
        fill="#fff"
      />
    </svg>
  </div>
</body>
`;
